File: angular6-portail-message/src/app/message/message.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpHeaders } from '@angular/common/http';

import { Message } from './message';
import { Observable } from 'rxjs';

const httpOptions = {
  headers: new HttpHeaders({ 'Content-Type': 'application/json' })
};

@Injectable()
export class MessageService {

  constructor(private http: HttpClient) { }

  private messageUrl = 'http://localhost:8080/message'

   public getAll(): Observable<Message[]> {
    console.log("message.service::getAll");
    return this.http.get<Message[]>(this.messageUrl);
  }
  
  public getOneById(id: number): Observable<Message> {
    console.log("message.service::get Message with id=" + id);
    const url = `${this.messageUrl}/${id}`;
    return this.http.get<Message>(url);
  }

  public delete(id: number): Observable<Message> {
    console.log("message.service::get Message with id=" + id);
    const url = `${this.messageUrl}/${id}`;
    return this.http.delete<Message>(url);
  }
}

File: java-portail-message/src/main/java/co/simplon/portail/message/service/MessageService.java
package co.simplon.portail.message.service;

import java.util.List;

import co.simplon.portail.message.model.Message;

public interface MessageService {
	
	List<Message> getAll();

	Message getOneById(long id);

	Message create(Message message);

	Message update(Message message);

	void delete(Message message);

	List<Message> getAllTourMessages(long id);

}
